### PR TITLE
fixed tests and run pytest

### LINKED ISSUE #270: Fix Tests
Tests need to be updated.

### LINKED ISSUE #273: Adding unittest to p5py pipeline
We need to add Python Unit Test app to the pipeline. 

https://github.com/marketplace/actions/python-unit-test

### PR DESCRIPTION
**Pull request by tushar5526**
Fixes #270 
Fixes #273 
Tests are working fine now, IO and Data module do not have any tests, and updated pylinter to include tests now.

### CODE REVIEW

File: p5/core/tests/test_shape.py [hunk 1]
```diff
@@ -106,40 +106,42 @@ def test_transforms(self):
...
-                           np.array([[1, 0], [2, 1], [3, 2], [0, 3], [4, 5], [6, 5], [7, 6], [4, 7]])))
-
-        self.assertTrue(
-            np.array_equal(contour_shape._tri_faces,
-                           np.array([[0, 4, 7], [4, 0, 1], [3, 7, 6], [7, 3, 0], [5, 1, 2], [1, 5, 4], [6, 2, 3], [2, 6, 5]])))
-
-        self.assertTrue(
-            np.array_equal(contour_shape._tri_vertices,
-                           np.array([[0.0, 0.0], [100.0, 0.0], [100.0, 100.0], [0.0, 100.0], [25.0, 25.0], [75.0, 25.0], [75.0, 75.0], [25.0, 75.0]])))
+    # def test_edges(self):
```
Comment (parsoyaarihant) [inline p5/core/tests/test_shape.py:109]: Is there any reason for commenting these functions?

Comment (tushar5526) [reply p5/core/tests/test_shape.py:109]: There are no members like `_tri_edges, _retriangulate, _compute_outline_edges` in Pshape or "quad". So for now I have commented out these tests and will try to get these tests fixed later, after figuring out some work around.

File: p5/core/tests/test_geometry.py [hunk 2]
```diff
@@ -54,10 +58,10 @@ def test_vertices(self):
                                                     0.5, -0.5], [-0.5, 0.5, 0.5],
             [0.5, -0.5, -0.5], [0.5, 0.5, -0.5], [0.5, -0.5, 0.5], [0.5, 0.5, 0.5],
             [-0.5, -0.5, -0.5], [0.5, -0.5, -
-                                 0.5], [-0.5, -0.5, 0.5], [0.5, -0.5, 0.5],
+            0.5], [-0.5, -0.5, 0.5], [0.5, -0.5, 0.5],
```
Comment (nakul-shahdadpuri) [inline p5/core/tests/test_geometry.py:61]: could you add comments for which vertex represents which corner of a cube ? Could improve code readability.

Comment (tushar5526) [reply p5/core/tests/test_geometry.py:61]: Yes sure, a lot of code is not documented in p5py, but I will add the comments after my internals :/

**Comment by tushar5526** [comment]
Doing `pytest -sv .` in `pylint.yml` is failing the build, but it is working fine on my system. Will look into this soon.

File: p5/core/tests/test_shape.py (hunk 1 above)
Comment (parsoyaarihant) [reply p5/core/tests/test_shape.py:109]: You can remove these then tests then.

Comment (parsoyaarihant) [reply p5/core/tests/test_shape.py:109]: I meant you can delete the lines which you have commented

File: .github/workflows/pylint.yml [hunk 3]
```diff
@@ -18,7 +18,11 @@ jobs:
...
+        pip install pytest
+    - name: Run pylint
       run: |
-        pylint --disable=I --disable=W --disable=C --disable=R `find . -type d \( -path ./p5/core/tests -o -path ./profiling -o -path ./p5/sketch/tests -o -path ./p5/pmath/tests \) \
-        -prune -false -o -name '*.py'`
+        pylint --disable=I --disable=W --disable=C --disable=R `find .  -path ./profiling -prune -false -o -name '*.py'`
+    - name: Run Tests
+      run: |
+        sudo apt-get install python-opengl libglfw3 -y
+        pytest -svv p5
```
Comment (parsoyaarihant) [inline .github/workflows/pylint.yml:28]: What does the -svv tag do?

Comment (tushar5526) [reply .github/workflows/pylint.yml:28]: These tags just modify the output.
*pytest -v .* gives this (-v verbose )
```
p5/core/tests/test_color.py::TestColor::test_parser PASSED               [  2%]
p5/core/tests/test_geometry.py::TestBoxGeometry::test_normals PASSED     [  5%]
p5/core/tests/test_geometry.py::TestBoxGeometry::test_triangle_edges PASSED [  8%]
p5/core/tests/test_geometry.py::TestBoxGeometry::test_vertices PASSED    [ 10%]
p5/core/tests/test_shape.py::TestPShape::test_properties PASSED          [ 13%]
p5/core/tests/test_shape.py::TestPShape::test_transforms PASSED          [ 16%]
p5/core/tests/test_svg.py::TestSVG::test_get_style PASSED                [ 18%]
p5/core/tests/test_svg.py::TestSVG::test_parse_line PASSED               [ 21%]
p5/core/tests/test_svg.py::TestSVG::test_parse_rect PASSED               [ 24%]
p5/pmath/tests/test_curves.py::TestCurves::test_bezier_point PASSED      [ 27%]
p5/pmath/tests/test_curves.py::TestCurves::test_bezier_tangent PASSED    [ 29%]
p5/pmath/tests/test_curves.py::TestCurves::test_curve_point PASSED       [ 32%]
p5/pmath/tests/test_curves.py::TestCurves::test_curve_tangent PASSED     [ 35%]
p5/pmath/tests/test_curves.py::TestCurves::test_quadratic_point PASSED   [ 37%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_look_at PASSED           [ 40%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_magnitude PASSED         [ 43%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_normalize PASSED         [ 45%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_perspective_matrix PASSED [ 48%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_rotation_matrix PASSED   [ 51%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_scale_transform PASSED   [ 54%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_translation_matrix PASSED [ 56%]
p5/pmath/tests/test_matrix.py::TestMatrix::test_triple_axis_rotation_matrix PASSED [ 59%]
p5/pmath/tests/test_utils.py::TestUtils::test_constrain PASSED           [ 62%]
p5/pmath/tests/test_utils.py::TestUtils::test_distance PASSED            [ 64%]
p5/pmath/tests/test_utils.py::TestUtils::test_lerp PASSED                [ 67%]
p5/pmath/tests/test_utils.py::TestUtils::test_magnitude PASSED           [ 70%]
p5/pmath/tests/test_utils.py::TestUtils::test_normalize PASSED           [ 72%]
p5/pmath/tests/test_utils.py::TestUtils::test_remap PASSED               [ 75%]
p5/pmath/tests/test_utils.py::TestUtils::test_sq PASSED                  [ 78%]
p5/pmath/tests/test_vector.py::TestVector::test_angle PASSED             [ 81%]
p5/pmath/tests/test_vector.py::TestVector::test_angle_between PASSED     [ 83%]
p5/pmath/tests/test_vector.py::TestVector::test_arithmetic PASSED        [ 86%]
p5/pmath/tests/test_vector.py::TestVector::test_cross PASSED             [ 89%]
p5/pmath/tests/test_vector.py::TestVector::test_distance PASSED          [ 91%]
p5/pmath/tests/test_vector.py::TestVector::test_dot PASSED               [ 94%]
p5/pmath/tests/test_vector.py::TestVector::test_lerp PASSED              [ 97%]
p5/pmath/tests/test_vector.py::TestVector::test_properties PASSED        [100%]
```
**pytest -sv .* give ( -s disables per-test capturing or print statements inside the code are not printed unless a test fails, increases visibility )
```
p5/core/tests/test_color.py::TestColor::test_parser PASSED
p5/core/tests/test_geometry.py::TestBoxGeometry::test_normals PASSED
p5/core/tests/test_geometry.py::TestBoxGeometry::test_triangle_edges PASSED
p5/core/tests/test_geometry.py::TestBoxGeometry::test_vertices PASSED
p5/core/tests/test_shape.py::TestPShape::test_properties PASSED
p5/core/tests/test_shape.py::TestPShape::test_transforms PASSED
p5/core/tests/test_svg.py::TestSVG::test_get_style PASSED
p5/core/tests/test_svg.py::TestSVG::test_parse_line PASSED
p5/core/tests/test_svg.py::TestSVG::test_parse_rect PASSED
p5/pmath/tests/test_curves.py::TestCurves::test_bezier_point PASSED
p5/pmath/tests/test_curves.py::TestCurves::test_bezier_tangent PASSED
p5/pmath/tests/test_curves.py::TestCurves::test_curve_point PASSED
p5/pmath/tests/test_curves.py::TestCurves::test_curve_tangent PASSED
p5/pmath/tests/test_curves.py::TestCurves::test_quadratic_point PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_look_at PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_magnitude PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_normalize PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_perspective_matrix PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_rotation_matrix PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_scale_transform PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_translation_matrix PASSED
p5/pmath/tests/test_matrix.py::TestMatrix::test_triple_axis_rotation_matrix PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_constrain PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_distance PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_lerp PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_magnitude PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_normalize PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_remap PASSED
p5/pmath/tests/test_utils.py::TestUtils::test_sq PASSED
p5/pmath/tests/test_vector.py::TestVector::test_angle PASSED
p5/pmath/tests/test_vector.py::TestVector::test_angle_between PASSED
p5/pmath/tests/test_vector.py::TestVector::test_arithmetic PASSED
p5/pmath/tests/test_vector.py::TestVector::test_cross PASSED
p5/pmath/tests/test_vector.py::TestVector::test_distance PASSED
p5/pmath/tests/test_vector.py::TestVector::test_dot PASSED
p5/pmath/tests/test_vector.py::TestVector::test_lerp PASSED
p5/pmath/tests/test_vector.py::TestVector::test_properties PASSED

```

**Comment by parsoyaarihant** [comment]
Can you separate the pylint and pytest as separate workflows?

**Comment by parsoyaarihant** [comment]
It will be more helpful for the user to know where the PR is failing.

**Comment by tushar5526** [comment]
> It will be more helpful for the user to know where the PR is failing.

Something like printing the logs here in conversations? I will add that too :)

**Comment by tushar5526** [comment]
> Can you separate the pylint and pytest as separate workflows?

Sure!

**Comment by parsoyaarihant** [comment]
> > It will be more helpful for the user to know where the PR is failing.
> 
> Something like printing the logs here in conversations? I will try that too :)

I meant if the user can see pylint and pytest separately, it will help him know where his PR is failing.

**Comment by tushar5526** [comment]
> > > It will be more helpful for the user to know where the PR is failing.
> > 
> > 
> > Something like printing the logs here in conversations? I will try that too :)
> 
> I meant if the user can see pylint and pytest separately, it will help him know where his PR is failing

Yes, I will do that.
It will be better that way.

**Pull request merged**